libcodechecker/analyze/log_parser.py

```diff
@@ -27,13 +28,84 @@
      '^-std=.*'])
 
 
-def get_compiler_includes(compiler, lang, compile_opts, extra_opts=None):
-    """
-    Returns a list of default includes of the given compiler.
-    """
+def get_compiler_err(cmd):
+    '''
+    Returns the stderr of a compiler invocation as string.
+    '''
+    try:
+        proc = subprocess.Popen(shlex.split(cmd),
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        _, err = proc.communicate("")
+        return err
+    except OSError as oerr:
+        LOG.error("Error during process execution: " + cmd + '\n' +
+                  oerr.strerror + "\n")
+
+
+def parse_compiler_includes(lines):
+    '''
+    Parse the compiler include paths from a string
+    '''
     start_mark = "#include <...> search starts here:"
     end_mark = "End of search list."
 
+    include_paths = []
+
+    do_append = False
+    for line in lines.splitlines(True):
+        line = line.strip()
+        if line.startswith(end_mark):
+            do_append = False
```

Comment:
Can there be multiple occurrences of start_mark? If no, could we just break here?

Reply:
This is the original code for parsing, unfortunately diff shows this as completely new code.
Still, I think we could just break here.